rs/boundary_node/ic_boundary/src/core.rs

```diff
@@ -860,6 +961,7 @@ pub fn setup_router(
         .merge(subnet_read_state_route)
         .merge(status_route)
         .merge(health_route)
+        .layer(Extension(salt))
```

Comment:
Ideally the non-per-request stuff should go into middlware State, not Extensions, to be type safe.

Since the salt is only used in metrics, then it probably should go into `HttpMetricParams` which is the metrics' middleware state.

Reply:
Should be good now 🙏